### ISSUE TITLE
Cannot get kubeconfig

### ISSUE THREAD

**Issue by stobias123** [kubernetes-sigs/cluster-api-provider-vsphere#163]
I'm using the OVA bootstrap cluster, and getting the below error from the `vsphere-provider-controller-manager` pod.

I've tried with both dhcp and static network types.. not sure what I'm missing. Seems to be an annotation?

Error from controller pod...
```
I1219 16:07:46.016757       1 clusteractuator.go:236] Master node [vs-master-2d8r6] IP not ready yet: could not get IP
I1219 16:07:46.027737       1 utils.go:159] cannot get kubeconfig because found no IP
```

**Comment by stobias123** [comment]
@figo That VM doesn't exist.

**Comment by stobias123** [comment]
@sidharthsurana 

Saw this bit
```
E1220 02:24:47.402524       1 machineactuator.go:67] path 'Resources' resolves to multiple resource pools
I1220 02:24:58.405836       1 clusteractuator.go:236] Master node [vs-master-2d8r6] IP not ready yet: could not get IP
I1220 02:24:58.504890       1 utils.go:159] cannot get kubeconfig because found no IP
```
Resolved via 
```
stobias $ govc find . -type p
  /LAB/host/LAB/Resources
  /LAB/host/LAB/Resources/RB6502
```

Now getting an error about disks.
```
I1220 02:28:18.735720       1 clusteractuator.go:118] [cluster-actuator] error retrieving kubeconfig for target cluster, will requeue
E1220 02:28:25.207350       1 machineactuator.go:67] [FATAL] None of the disks specified in the MachineSpec matched with the disks on the template ubuntu-16.04-server-cloudimg-amd64
```
Is the govmomi code not applying customization?

**Comment by stobias123** [comment]
Deleted  `disks` block, and made sure to specify full `govc` paths to objects, I was then able to have the master provisioned.

**Comment by frapposelli** [comment]
@stobias123 were you able to solve this?

/priority awaiting-more-evidence
/kind bug

**Comment by stobias123** [comment]
Running off the latest version/build I was able to use the OVA properly. The master node is created, kubeconfig is pulled, but then basically nothing happens... Could be other issues. 

Running into some other possible issues related to machine objects, and multiple cluster resource objects. The bootstrap cluster seems to only pull from kubeconfig from the first created master. more testing to come on this.

tl;dr - this issue seems to be resolved, however will possibly open other issues depending on further testing.

**Comment by figo** [comment]
@stobias123 thanks for providing the feedback, i can confirm that creating a second cluster failed in my setup, please refer this issue: #191

**Comment by sidharthsurana** [comment]
@stobias123 based on your last comment, were you able to do further testing?
Can you confirm if you still encountering this issue?

**Comment by stobias123** [comment]
@sidharthsurana I'm unable to `make prod-yaml` in the latest code as of yesterday.

**Comment by sflxn** [comment]
@stobias123 There seems to be multiple problems mentioned in this issue.

In reference to your issue with `make prod-yaml`, we've merged in some fixes for the yaml generation in the last 10 days.  You must upgrade your kustomize version to 2.x.  Even with that version, the repo had an issue with kustomize but we merged in some fixes recently, and it should work now.  Can you verify?

In reference to your static/dhcp issue, you mentioned dhcp works but not static.  Is it possibly related to https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/185?

As for multiple master nodes, yes, if you're using clusterctl, we do not yet have a way to pull kubeconfig from the other master nodes yet.  However, you can still do it manually.  Your generated keys should be in ~/.ssh/.  If you use `ssh -i vsphere_tmp <master>`, you can get onto that master node.  The kubeconfig file should be under /etc/kubernetes/admin.conf.

Lastly, can you provide snippets of the yaml to show us how you're attempting to create multiple masters?
